### PR TITLE
Rework basic 2D navigation demo for Godot 3.5 and later

### PR DESCRIPTION
**Pull request by smix8**
Reworks basic 2D navigation demo to use new NavigationServer and NavigationAgent from Godot 3.5 / 4.0.

Keeps the demos scope to basic navigation paths and movement, so no avoidance or physics.

Fixes tutorial issue from https://github.com/godotengine/godot-docs/issues/6055

The demo is prepared for Godot 3.5 but I did not find a 3.5 branch so I picked master.
For Godot 4.0 port change 'Navigation2DServer' to 'NavigationServer2D' and export/onready to new syntax.

### CODE REVIEW

**Comment by Calinou** [comment]
> The demo is prepared for Godot 3.5 but I did not find a 3.5 branch so I picked master.

The `master` branch of godot-demo-projects is indeed for Godot 3.x :slightly_smiling_face: 

The `4.0-dev` branch is where 4.0 demos are stored.

**Comment by krupitskas** [comment]
@smix8 can you retarget merge branch to 4.0-dev please?

**Comment by Calinou** [comment]
> @smix8 can you retarget merge branch to 4.0-dev please?

This PR should keep targeting `master`, but a new PR targeting `4.0-dev` can be created. This would be useful as I haven't gotten the demo to work successfully in https://github.com/godotengine/godot-demo-projects/pull/782.

**Comment by idbrii** [comment]
This is great! I used the navmesh demo from master for my project and then was very confused how to apply navagents to it, so having this simpler demo instead would be a great help.

I tested out this PR on 3.5.1-stable and it works wonderfully.

Reviewing the code, my only comment is to restore the comment explaining where "click" comes from:

```patch
--- a/2d/navigation/navigation.gd
+++ b/2d/navigation/navigation.gd
@@ -4,6 +4,8 @@ extends Node2D
 onready var character = $Character
 
 
+# The "click" event is a custom input action defined in
+# Project > Project Settings > Input Map tab.
 func _unhandled_input(event):
 	if not event.is_action_pressed("click"):
 		return
```

Or rework it to hardcode the mouse button like the 3d example: `event is InputEventMouseButton and event.button_index == BUTTON_LEFT and event.pressed`.

**Comment by Calinou** [comment]
@smix8 Could you rebase this PR against `master` (for 3.5.1)? I've tested the demo locally and it works great.

One thing I'd change though is the navigation path color – the current blue color is hard to see. I'd try an orange color instead.

**Comment by smix8** [comment]
Changed line color to red cause that is the path debug color used by default in Godot 4 as well.

File: 2d/navigation/character.gd [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		target_position,
+		optimize
+		)
+
+
+func _physics_process(delta):
+	if navigationagent.is_navigation_finished():
+		return
+	var next_position : Vector2 = navigationagent.get_next_location()
+	global_position = global_position.move_toward(next_position, delta * character_speed)
```
Comment (Calinou) [inline 2d/navigation/character.gd:34]:
```suggestion
extends Sprite

export(float) var character_speed = 400.0

onready var navigation_agent: NavigationAgent2D = $NavigationAgent2D
onready var navigation_path_debug_line: Line2D = $Line2D


func _ready():
	navigation_path_debug_line.set_as_toplevel(true)


func set_target_location(target_location: Vector2):
	navigation_agent.set_target_location(target_location)
	# Alternative way to get a full navigation path with Server API.
	var navigation_map: RID = get_world_2d().get_navigation_map()
	var start_position: Vector2 = get_global_transform().get_origin()
	var target_position: Vector2 = target_location
	var optimize: bool = true
	navigation_path_debug_line.points = Navigation2DServer.map_get_path(
			navigation_map,
			start_position,
			target_position,
			optimize
	)


func _physics_process(delta):
	if navigation_agent.is_navigation_finished():
		return

	var next_position: Vector2 = navigation_agent.get_next_location()
	global_position = global_position.move_toward(next_position, delta * character_speed)
```

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by akien-mga** [comment]
This needs to be forward ported to 4.0 in the `4.0-dev` branch.

**Pull request merged**